Jenkinsfile

```diff
@@ -0,0 +1,357 @@
+// -*- mode: groovy -*-
+// vim: set filetype=groovy :
+
+// Required plugins
+// - Pipeline
+// - Pipeline: Multibranch
+// - HTTP Request Plugin
+// - Stash Notifier
+// - Timestamper
+
+properties( [
+  disableConcurrentBuilds(),
+  [ $class: 'BuildDiscarderProperty', strategy: [ $class: 'LogRotator', daysToKeepStr: '25' ] ],
+] )
+
+podLabel = "react-dnd-ax-${UUID.randomUUID()}"
+
+lastSuccessfulCommit = getLastSuccessfulCommit()
+currentCommit = commitHashForBuild(currentBuild.rawBuild)
+
+podTemplate(
+  label: podLabel,
+  serviceAccount: env.KUBERNETES_SERVICE_ACCOUNT,
+  annotations: [
+    podAnnotation(key: 'iam.amazonaws.com/role', value: 'arn:aws:iam::392477962641:role/learn-ci-jenkins-agent')
+  ],
+  containers: [
+    containerTemplate(name: 'build', image: 'node:8', ttyEnabled: true, command: '/bin/cat')
+  ],
+  volumes: [
+    secretVolume( secretName: 'npm-publish-token', mountPath: '/home/jenkins/npm-publish-token' ),
+    secretVolume( secretName: 'stash-jenkins-nirvana', mountPath: '/home/jenkins/ssh-stash-jenkins-nirvana' ),
+  ]){
+  // common pipeline template
+  try {
+    withSlaveAndEnvironment(podLabel) {
+      // begin custom pipeline steps
+      container('build') {
+        gitCheckout(true)
+        withStashCredentials {
+          withNotifications {
+            prepare()
+            unitTest()
+            build()
+            if(isProductionBranch() && !triggeredByVersionBump()) {
+              publishToRepo()
+            }
+          }
+        }
+      }
+    }
+  }
+  catch (error) {
+      echo "Error running pipeline, stack follows"
+      error.printStackTrace()
+  }
+}
+
+// Begin common pipeline template methods
+def withSlaveAndEnvironment(String label, Closure block) {
+  node(label) {
+    timestamps {
+      block()
+    }
+  }
+}
+
+def withStashCredentials( Closure block ) {
+  try {
+    def name = 'Jenkins'
+    def email = "jenkins@blackboard.com"
+
+    // Temporary activate the credentials volume and set up git
+    sh """#!/usr/bin/env bash
+      set -ex
+
+      rm -rf /root/.ssh
+
+      cp -R /home/jenkins/ssh-stash-jenkins-nirvana /root/.ssh
+
+      chmod 400 /root/.ssh/id_rsa
+
+      git config --global user.name $name
+      git config --global user.email $email
+    """
+
+    def envVars = [
+      "GIT_AUTHOR_NAME=$name",
+      "GIT_AUTHOR_EMAIL=$email",
+      "GIT_COMMITTER_NAME=$name",
+      "GIT_COMMITTER_EMAIL=$email",
+      "NPM_CONFIG_LOGLEVEL=warn"
+    ]
+    withEnv( envVars ) {
+      block()
+    }
+  } finally {
+    sh 'rm -rf /root/.ssh'
+  }
+}
+
+//
+// Build stage methods
+//
+def gitCheckout( showInChangelog ) {
+  checkout changelog: showInChangelog, scm: [
+    $class: 'GitSCM',
+    branches: [ [ name: env.BRANCH_NAME ] ],
+    browser: [ $class: 'Stash', repoUrl: 'https://stash.bbpd.io/projects/SHARED/repos/react-dnd-ax' ],
```

Comment:
Echoing David's comment the jenkinsfile includes a lot of stuff related to it being in shared which we no longer need.